### ISSUE TITLE
Can't log arrays (TypeError)

### ISSUE THREAD

**Issue by hex13** [mweststrate/immer#50]
immer 0.4.1, Node v8.4.0

```javascript
const produce = require('immer').default;
produce([],(arr) => { console.log(arr);  });
```

```
TypeError: 'getOwnPropertyDescriptor' on proxy: trap returned descriptor for property 'length' that is incompatible with the existing property in the proxy target
    at Function.keys (<anonymous>)
    at formatValue (util.js:392:21)
    at inspect (util.js:236:10)
    at format (util.js:97:24)
    at Console.log (console.js:113:24)
    at immer (repl:1:28)
    at produce (blah/node_modules/immer/immer.js:224:29)
    at repl:1:1
    at ContextifyScript.Script.runInThisContext (vm.js:44:33)
    at REPLServer.defaultEval (repl.js:239:29)
```

**Comment by zhw2590582** [comment]
https://github.com/mweststrate/immer/blob/29e22580849f66311d089a3362972053b1ecbe09/immer.js#L157
When you directly access an array itself, 'length' property is an exception from this, immer attempts to modify the descriptor's configurable property, So TypeError.
How is this?
```js
if (descriptor && (Array.isArray(owner) && prop !== 'length')) descriptor.configurable = true
```
@mweststrate

**Comment by mweststrate** [comment]
@zhw2590582 correct! Forgot that `Array.length` is as usual an exception to everything :/. working on a fix.

**Comment by mweststrate** [comment]
Should be fixed in 0.4.2
